eth/client.go

```diff
@@ -185,7 +186,7 @@ func (c *PooledClient) NonceAt(ctx context.Context, account common.Address) (uin
 func (c *PooledClient) SendTransaction(ctx context.Context, tx *types.Transaction) (SendTransactionResponse, error) {
 	data, err := rlp.EncodeToBytes(tx)
 	if err != nil {
-		return SendTransactionResponse{}, err
+		return SendTransactionResponse{}, errors.Wrap(err, "Failed to encode transaction")
```

Comment:
Does the `errors` lib allow us to easily include meaningful context? For example here, `tx`, or in line 350 ("Failed to dia websocket"), the `url`. If yes, let's use it. If not, it would still be useful to include (potentially more limited) context in the message itself, with `fmt.Sprintf()`.

Reply:
Yep, the `github.com/pkg/errors` library automatically includes the stack trace when it wraps the error. Can definitely also include more context in the error message itself as well